.github/workflows/tests.yml

```diff
@@ -0,0 +1,36 @@
+name: Tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+env:
+  HATCH_VERSION: "1.14.0"
+  OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    strategy:
+      max-parallel: 1
+      matrix:
+        python-version: [3.9, 3.10.x, 3.11, 3.12]
```

Comment:
why 3.10.x?

(I may not be familiar with this syntax)

Reply:
Because of [this](https://github.com/actions/runner/issues/1989) mainly (3.10 is treated at 3.1).

But I found that you can also do `'3.10'` (as of [this](https://github.com/actions/runner/issues/1989)), will update accordingly.